app/Filament/Resources/UserResource/Pages/ListUsers.php

```diff
@@ -167,4 +178,39 @@ protected function getHeaderActions(): array
             CreateAction::make(),
         ];
     }
+
+    protected function getRolesOption(): array
+    {
+        $roles = Role::query()->take(50)->pluck('name', 'id')->toArray();
+
+        return [
+            '' => [
+                'none' => 'None',
+            ],
+            'Roles' => [
+                ...$roles,
+            ],
+        ];
+    }
+
+    protected function roleFilter(Builder $query, array $data): void
+    {
+        if (empty($data['values'])) {
+            return;
+        }
+
+        $query->where(function ($query) use ($data) {
```

Comment:
Does the `->where()` here do anything? Filament will add it around each filter out of the box for you I think